algebird-core/src/main/scala/com/twitter/algebird/Max.scala

```diff
@@ -58,51 +60,55 @@ object Max extends MaxInstances {
   /**
    * Returns a [[Semigroup]] instance with a `plus` implementation
    * that always returns the maximum `T` argument.
-   *
-   * @todo Note that this is a semilattice, for when we merge  typelevel/algebra.
    */
-  def maxSemigroup[T](implicit ord: Ordering[T]): Semigroup[T] =
-    Semigroup.from { (l: T, r: T) => ord.max(l, r) }
+  def maxSemigroup[T](implicit ord: Ordering[T]): Semigroup[T] with Semilattice[T] =
+    new Semigroup[T] with Semilattice[T] {
+      def plus(l: T, r: T): T = ord.max(l, r)
+    }
 }
 
 private[algebird] sealed abstract class MaxInstances {
   implicit def equiv[T](implicit eq: Equiv[T]): Equiv[Max[T]] = Equiv.by(_.get)
   implicit def ordering[T: Ordering]: Ordering[Max[T]] = Ordering.by(_.get)
 
-  private[this] def plus[T](implicit ord: Ordering[T]) = {
-    (l: Max[T], r: Max[T]) => if (ord.gteq(l.get, r.get)) l else r
-  }
-
   /**
    * Returns a [[Monoid]] instance for [[Max]][T] that combines
    * instances using [[Max.max]] and uses `zero` for its identity.
    *
    * @param zero identity of the returned [[Monoid]] instance
    * @note `zero` must be `<=` every element of `T` for the returned instance to be lawful.
    */
-  def monoid[T: Ordering](zero: => T): Monoid[Max[T]] = Monoid.from(Max(zero))(plus)
+  def monoid[T: Ordering](z: => T): Monoid[Max[T]] with BoundedSemilattice[Max[T]] =
```

Comment:
then whe we override `zero` below you have the chance for confusion (in fact, you can create an infinite loop if you are not careful).